> What is the result of CI when you remove these?
> 
> Could you add a link to Azure for a sample run?

https://dev.azure.com/DeeDeeG/b/_build/results?buildId=953

This run has just this branch's changes rebased on top of `master`, because the Electron 9 branch isn't building right now, unrelatedly to this PR.

---

I also did a bunch of runs with the startup listener thing enabled during testing. See here: [CI link](https://dev.azure.com/DeeDeeG/b/_build?definitionId=17&branchFilter=113) (Runs with commit hash d13cc88.)

---

I see this error in most (but not all) runs, so that needs looking into:

```
##[error] Tests for core-render-process in D:/a/1/s/spec/pane-spec.js. failed.

Pane
  pending state
    it Pane pending state does not open file in pending state by default
      timeout: timed out after 120000 msec waiting for spec promise to resolve
```

--- a/spec/integration/helpers/start-atom.js
+++ b/spec/integration/helpers/start-atom.js
@@ -59,9 +59,6 @@ const buildAtomClient = async (args, env) => {
   // So, for now, we'll just use a random port (we don't care about its value since we're not
   // connecting through it).
   // (inspired by https://github.com/electron/spectron/pull/361/commits/737db138bd8a6daaf80f9c2bff710ce4a5fff39b).
-  // TodoElectronIssue: Remove the whole remote-debugging-port param once we upgrade
-  // to Electron v5, since this was fixes there (see electron/electron#17800).
-  const randomPort = Math.floor(Math.random() * (9999 - 9000) + 9000);
 
   const userDataDir = temp.mkdirSync('atom-user-data-dir');
   const client = await webdriverio.remote({
@@ -80,7 +77,6 @@ const buildAtomClient = async (args, env) => {
           'dev',
           'safe',
           `user-data-dir=${userDataDir}`,
-          `remote-debugging-port=${randomPort}`
         ]
       }
     }

--- a/spec/integration/helpers/start-atom.js
+++ b/spec/integration/helpers/start-atom.js
@@ -53,13 +53,6 @@ const chromeDriverDown = done => {
 };
 
 const buildAtomClient = async (args, env) => {
-  // Since ChromeDriver v2.41, ChromeDriver will only connect if, either we precise a port
-  // for remote debugging, either the embedder (ie electron) made sure to pass `USER_DATA_DIR`
-  // to the remote debugging server.
-  // So, for now, we'll just use a random port (we don't care about its value since we're not
-  // connecting through it).
-  // (inspired by https://github.com/electron/spectron/pull/361/commits/737db138bd8a6daaf80f9c2bff710ce4a5fff39b).
-
   const userDataDir = temp.mkdirSync('atom-user-data-dir');
   const client = await webdriverio.remote({
     host: 'localhost',

--- a/src/main-process/atom-application.js
+++ b/src/main-process/atom-application.js
@@ -102,17 +102,7 @@ const createSocketSecret = async atomVersion => {
 
 const encryptOptions = (options, secret) => {
   const message = JSON.stringify(options);
-
-  // Even if the following IV is not cryptographically secure, there's a really good chance
-  // it's going to be unique between executions which is the requirement for GCM.
-  // We're not using `crypto.randomBytes()` because in electron v2, that API is really slow
-  // on Windows machines, which affects the startup time of Atom.
-  // TodoElectronIssue: Once we upgrade to electron v3 we can use `crypto.randomBytes()`
-  const initVectorHash = crypto.createHash('sha1');
-  initVectorHash.update(Date.now() + '');
-  initVectorHash.update(Math.random() + '');
-  const initVector = initVectorHash.digest();
-
+  const initVector = crypto.randomBytes(16); // AES uses 16 bytes for iV
   const cipher = crypto.createCipheriv('aes-256-gcm', secret, initVector);
 
   let content = cipher.update(message, 'utf8', 'hex');

